File: my-code/167/main.go
package main

import "fmt"

// @Time     : 2021/5/28 23:43
// @Author   : Ranshi
// @File     : 167. 两数之和 II - 输入有序数组.go

func twoSum(ints []int, target int) []int {
	left, right := 0, len(ints)-1
	for left < right {
		val := ints[left] + ints[right]
		if val == target {
			return []int{left + 1, right + 1}
		} else if val > target {
			right--
		} else {
			left++
		}
	}
	return nil
}

func main() {
	fmt.Println(twoSum([]int{2, 3, 4}, 6))
}

File: my-code/167/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2021/08/28 09:14
# @Author   : Ranshi
# @File     : main.py

from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        mapping = {}
        for i in range(len(numbers)):
            if numbers[i] in mapping:
                return [mapping[numbers[i]] + 1, i + 1]
            else:
                mapping[target - numbers[i]] = i
        return []


if __name__ == "__main__":
    print(Solution().twoSum(numbers=[2, 7, 11, 15], target=9))
